contracts/src/core/libraries/v3/PaymentVaultLib.sol

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {SafeERC20, IERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+
+library PaymentVaultTypes {
+    struct User {
+        uint256 deposit; // the total on demand deposit of the user
+        Reservation reservation;
+    }
+
+    struct Quorum {
+        QuorumProtocolConfig protocolCfg;
+        QuorumConfig cfg;
+        mapping(address => User) user;
+        mapping(uint64 => uint64) reservedSymbols; // reserved symbols per period in this quorum
+    }
+
+    struct QuorumConfig {
+        address token; // the address of the token used for on-demand payments.
+        address recipient; // the address of the recipient of the on-demand payments.
+        uint64 reservationSymbolsPerSecond;
+        uint64 onDemandSymbolsPerPeriod;
```

Comment:
let's update this to `onDemandSymbolsPerSecond` for consistency; the previous vault impl also had a misleading name:p